### PR TITLE
Adds storage policy to e2e tests

### PR DESCRIPTION
**Pull request by gab-satchi**
**What this PR does / why we need it**:
Adds e2e test that: 
- creates tag, category and storage policy
- creates machine template that utilizes the storage policy

**Release note**:
```release-note
None
```

### CODE REVIEW

File: test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml
Comment (yastij) [inline test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: is there a specific reason to have this file instead of using `packaging/flavorgen` ?

**Review by yastij — COMMENTED** [review]
will do a more complete pass tomorrow

File: test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml
Comment (gab-satchi) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: Not really, was a pattern when working with CAPA e2e. Just needed a 'flavour' without the VsphereMachineTemplate. Should I just use `MultiNodeTemplateWithHAProxy` and then take out machine template?

Comment (yastij) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: is there a reason why we don't need the `VSphereMachineTemplate` ?

Comment (gab-satchi) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: We definitely need it but the storage-policy name is determined at run time so the vsphereMachineTemplate is applied by the test once the storage-policy is set up.

Comment (yastij) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: Ideally we should just consume a storage policy that is already created in our environment, this way we can re-use the same one across jobs, without having to generate it at runtime and deal with the naming issues.

Comment (gab-satchi) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: ya could do that. Went with creating and cleaning up per test so the user would have less setup to do. And the validation becomes simpler when it's a new storage policy rather than an existing one (can just make sure at least 1 vm is using it rather than checking against the vm-id is in the list of objects using the policy).

With using an existing sp, we'd still need a new flavour that specifies the storage-policy in the machine template. Or do you think I should just modify the quick start to use storage policy as well? We should cover both scenarios with tests imo.

File: test/e2e/capv_quick_start_test.go
Comment (srm09) [inline test/e2e/capv_quick_start_test.go:138]: Call them with defer to make sure they get run at the end of the method. Since there is a return after a failure.

Also, even if the `pbmClient` fails to exist, we should enforce deletion of the profile.

```golang 
..
defer tagsManager.DeleteTag(ctx, &tag)
defer tagsManager.DeleteCategory(ctx, &cat)
```

File: test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml
Comment (srm09) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: I prefer the programmatic way of doing the setup for creating a storage policy, which makes sure that there is no extra setup that needs to be done before running the test.

**Review by srm09 — COMMENTED** [review]
minor nits around resource cleanup for tests

File: test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml
Comment (yastij) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml:1]: I'd rather add changes to `flavorgen` to generate both (sp-only, and sp with datastore) and use an existing storage policy. which would allow to get rid of the file + test against an existing storage policy

Comment (gab-satchi) [reply test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml]: removed file and added storage policy to haproxy flavour.

File: test/e2e/capv_quick_start_test.go
Comment (gab-satchi) [inline test/e2e/capv_quick_start_test.go:103]: the query for objects using the storage policy returns server UID along with object reference ID. The vsphereVM spec has the BIOS UID which is different so forced to fetch the vm again and compare object references. let me know if there's a simpler way I missed

**Comment by gab-satchi** [comment]
need to rebase on top of our main branch 

/hold

**Review by yastij — APPROVED** [review]
/approve
/lgtm

**Pull request merged**
